### PR TITLE
Owen's Challenge Submission

### PR DESCRIPTION
**Pull request by owenchak**
For my submission, I used JS, Jquery and CSS since I didn't see the need here for any additional frameworks or preprocessors. I did considered using AngularJS for the two way data binding but JS was just as effective for this project. 

Note:
- To run, npm install -> node server
- Did not include unit tests

### CODE REVIEW

File: hash.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+function Hash()
```
Comment (NicHaley) [inline hash.js]: Instead of using a function constructor, it would have been nice you make use of ES6 Classes instead

File: index.html [hunk 2]
```diff
@@ -1,14 +1,32 @@
...
+    src="https://code.jquery.com/jquery-3.2.1.min.js"
+    integrity="sha256-hwg4gsxgFZhOsEEamdOYGBf13FyQuiTwlAQgxVSNgt4="
+    crossorigin="anonymous"></script>
+    <link href="https://fonts.googleapis.com/css?family=Roboto" rel="stylesheet">
+    <link rel="stylesheet" type="text/css" href="app.css">
+
+</head>
+<body>
+    <div class="input-container">
+        <input id="todo-input" type="text" placeholder="Todo: Hire Owen Chak" autofocus>
```
Comment (NicHaley) [inline index.html]: Nice placeholder ;)

File: client.js [hunk 3]
```diff
@@ -1,32 +1,164 @@
 const server = io('http://localhost:3003/');
+
+
 const list = document.getElementById('todo-list');
 
-// NOTE: These are all our globally scoped functions for interacting with the server
+var titleHashTable = new Hash();
+var hashIndexCounter=0;
+var offlineAddedTodos = [];
```
Comment (davidhariri) [inline client.js]: What was this for?

File: client.js [hunk 4]
```diff
@@ -1,32 +1,164 @@
...
+    var localStorageStringArray = [];
+    var index=0;
+    todos.forEach((todo) => {
+        //hash table -> searching for title = O(1)
+        if(todos.length>0){
+            titleHashTable.setItem(todo.title, hashIndexCounter);
+            hashIndexCounter++;
+
+            //push each todo item to localstorage in case of disconnect
+            localStorageStringArray.push(
```
Comment (davidhariri) [inline client.js]: reading between the lines here... I think you started building an offline mode and then bailed. Am I right about that? If so, I think that was a mistake- I'll talk about why in my final review notes at the end.

**Pull request closed without merging**